hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1009,59 +836,14 @@ def cached_test_function(self, buffer):
         Otherwise we call through to ``test_function``, and return a
         fresh result.
         """
-        rewritten = bytearray()
-        would_overrun = False
-
-        node_index = 0
-        for i, c in enumerate(buffer):
-            # If there's a forced value or a mask at this position, then
-            # pretend that the buffer already contains a matching value,
-            # because the test function is going to do the same.
-            try:
-                c = self.forced[node_index]
-            except KeyError:
-                pass
-            try:
-                c = c & self.masks[node_index]
-            except KeyError:
-                pass
-
-            try:
-                # If we know how many bytes are read at this point and
-                # there aren't enough, then it doesn't actually matter
-                # what the values are, we're definitely going to overrun.
-                if i + self.block_sizes[node_index] > len(buffer):
-                    would_overrun = True
-                    break
-            except KeyError:
-                pass
-
-            rewritten.append(c)
-
-            try:
-                node_index = self.tree[node_index][c]
-            except KeyError:
-                # The byte at this position isn't in the tree, which means
-                # we haven't tested this buffer. Break out of the tree
-                # traversal, and run the test function normally.
-                break
-            node = self.tree[node_index]
-            if isinstance(node, ConjectureData):
-                # This buffer (or a prefix of it) has already been tested.
-                # Return the stored result instead of trying it again.
-                assert node.status != Status.OVERRUN
-                return node
-        else:
-            # Falling off the end of this loop means that we're about to test
-            # a prefix of a previously-tested byte stream, so the test would
-            # overrun.
-            would_overrun = True
-
-        if would_overrun:
-            return Overrun
+        try:
+            return self.tree.cached_test_function(buffer)
+        except KeyError:
```

Comment:
I'm worried that a future revision of `tree.cached_test_function` might accidentally throw a `KeyError` due to an internal bug, which would then be silently masked by this catch block.

Would it be safer to throw a dedicated exception, or to return some other sentinel value instead?

Reply:
Yeah, that's a fair point. How would you feel about it returning `None` in this case?